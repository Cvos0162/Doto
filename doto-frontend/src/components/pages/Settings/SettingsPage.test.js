import React from "react";
import ReactDOM from "react-dom";
import { mount } from "enzyme";
import { BrowserRouter as Router } from "react-router-dom";
import SettingsPage from "./SettingsPage";
import { ThemeContext } from "../../../context/ThemeContext";

describe("<SettingsPage /> component being rendered", () => {
    let theme;
    const setTheme = jest.fn();
    const useStateSpy = jest.spyOn(React, "useState");
    useStateSpy.mockImplementation(theme => [theme, setTheme]);

    let subject;

    const Wrapper = () => {
        return (
            <Router>
                <ThemeContext.Provider value={[theme, setTheme]}>
                    <SettingsPage />
                </ThemeContext.Provider>
            </Router>
        );
    };

    beforeEach(() => {
        useStateSpy.mockImplementation(theme => [theme, setTheme]);

        subject = mount(<Wrapper />);
    });

    afterEach(() => {
        jest.clearAllMocks();
        subject.unmount();
    });

    it("SettingsPage component rendered without crashing", () => {
        const div = document.createElement("div");
        ReactDOM.render(<Wrapper />, div);
    });

    it("Make sure render matches snapshot", () => {
        expect(subject).toMatchSnapshot();
    });
});
