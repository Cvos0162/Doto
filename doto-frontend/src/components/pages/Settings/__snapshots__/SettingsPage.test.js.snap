// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SettingsPage /> component being rendered Make sure render matches snapshot 1`] = `ReactWrapper {}`;
